File: src/gatling/scala/org/openapitools/client/model/Job.scala

package org.openapitools.client.model


case class Job (
    /* Unique job identifier */
    _jobId: Option[String],
    /* Batch ID associated with the job (useful for linking with record creation requests) */
    _batchId: Option[String],
    /* State of the job (e.g., completed, pending) */
    _state: Option[String],
    /* Timestamp when the job was created */
    _dateCreated: Option[String],
    /* Timestamp when the job started processing */
    _dateStarted: Option[String],
    /* Timestamp when the job finished processing */
    _dateFinished: Option[String]
)
object Job {
    def toStringBody(var_jobId: Object, var_batchId: Object, var_state: Object, var_dateCreated: Object, var_dateStarted: Object, var_dateFinished: Object) =
        s"""
        | {
        | "jobId":$var_jobId,"batchId":$var_batchId,"state":$var_state,"dateCreated":$var_dateCreated,"dateStarted":$var_dateStarted,"dateFinished":$var_dateFinished
        | }
        """.stripMargin
}

File: models/Job.js
const utils = require('../utils/utils');

module.exports = {
    fields: (prefix = '', isInput = true, isArrayChild = false) => {
        const {keyPrefix, labelPrefix} = utils.buildKeyAndLabel(prefix, isInput, isArrayChild)
        return [
            {
                key: `${keyPrefix}jobId`,
                label: `Unique job identifier - [${labelPrefix}jobId]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}batchId`,
                label: `Batch ID associated with the job (useful for linking with record creation requests) - [${labelPrefix}batchId]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}state`,
                label: `State of the job (e.g., completed, pending) - [${labelPrefix}state]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}dateCreated`,
                label: `Timestamp when the job was created - [${labelPrefix}dateCreated]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}dateStarted`,
                label: `Timestamp when the job started processing - [${labelPrefix}dateStarted]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}dateFinished`,
                label: `Timestamp when the job finished processing - [${labelPrefix}dateFinished]`,
                type: 'string',
            },
        ]
    },
    mapping: (bundle, prefix = '') => {
        const {keyPrefix} = utils.buildKeyAndLabel(prefix)
        return {
            'jobId': bundle.inputData?.[`${keyPrefix}jobId`],
            'batchId': bundle.inputData?.[`${keyPrefix}batchId`],
            'state': bundle.inputData?.[`${keyPrefix}state`],
            'dateCreated': bundle.inputData?.[`${keyPrefix}dateCreated`],
            'dateStarted': bundle.inputData?.[`${keyPrefix}dateStarted`],
            'dateFinished': bundle.inputData?.[`${keyPrefix}dateFinished`],
        }
    },
}
